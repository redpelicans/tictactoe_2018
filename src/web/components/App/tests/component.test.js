import React from 'react';
import { shallow } from 'enzyme';
import Component from '..';

const player = { name: 'NAME' };
const computer = { name: 'computer', isComputer: true };
const board = ['x', 'o', 'x', null, null, 'x', 'o', null, null];
const history = [{ id: 1, winner: player }];

const state = {
  currentPlayer: player,
  status: 'GAME_OVER',
  player,
  computer,
  board,
  history,
};

describe('App', () => {
  it('should render', () => {
    const wrapper = shallow(<Component {...state} />);
    expect(wrapper).toMatchSnapshot();
  });
});
