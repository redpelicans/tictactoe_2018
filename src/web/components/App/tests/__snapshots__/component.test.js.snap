// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    board={
      Array [
        "x",
        "o",
        "x",
        null,
        null,
        "x",
        "o",
        null,
        null,
      ]
    }
    computer={
      Object {
        "isComputer": true,
        "name": "computer",
      }
    }
    currentPlayer={
      Object {
        "name": "NAME",
      }
    }
    history={
      Array [
        Object {
          "id": 1,
          "winner": Object {
            "name": "NAME",
          },
        },
      ]
    }
    player={
      Object {
        "name": "NAME",
      }
    }
    status="GAME_OVER"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": Array [
        <Unknown
          player={
            Object {
              "name": "NAME",
            }
          }
        >
          <Unknown>
            <Unknown
              type="trophy"
            />
            <Unknown
              name="TicTacToe"
            />
          </Unknown>
          <Unknown>
            <StartButton
              status="GAME_OVER"
            >
              Start The Game
            </StartButton>
          </Unknown>
        </Unknown>,
        <Jumbotron
          bsClass="jumbotron"
          className="content"
          componentClass="div"
        >
          <Grid
            bsClass="container"
            componentClass="div"
            fluid={false}
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={12}
              >
                <PiePanel
                  history={
                    Array [
                      Object {
                        "id": 1,
                        "winner": Object {
                          "name": "NAME",
                        },
                      },
                    ]
                  }
                  player={
                    Object {
                      "name": "NAME",
                    }
                  }
                />
              </Col>
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={12}
              >
                <Unknown
                  board={
                    Array [
                      "x",
                      "o",
                      "x",
                      null,
                      null,
                      "x",
                      "o",
                      null,
                      null,
                    ]
                  }
                  computerPlay={[Function]}
                  currentPlayer={
                    Object {
                      "name": "NAME",
                    }
                  }
                />
              </Col>
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={12}
              >
                <Unknown
                  history={
                    Array [
                      Object {
                        "id": 1,
                        "winner": Object {
                          "name": "NAME",
                        },
                      },
                    ]
                  }
                />
              </Col>
            </Row>
          </Grid>
        </Jumbotron>,
      ],
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Unknown>
              <Unknown
                type="trophy"
              />
              <Unknown
                name="TicTacToe"
              />
            </Unknown>,
            <Unknown>
              <StartButton
                status="GAME_OVER"
              >
                Start The Game
              </StartButton>
            </Unknown>,
          ],
          "player": Object {
            "name": "NAME",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Unknown
                  type="trophy"
                />,
                <Unknown
                  name="TicTacToe"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "type": "trophy",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "name": "TicTacToe",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <StartButton
                status="GAME_OVER"
              >
                Start The Game
              </StartButton>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Start The Game",
                "status": "GAME_OVER",
              },
              "ref": null,
              "rendered": "Start The Game",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "jumbotron",
          "children": <Grid
            bsClass="container"
            componentClass="div"
            fluid={false}
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={12}
              >
                <PiePanel
                  history={
                    Array [
                      Object {
                        "id": 1,
                        "winner": Object {
                          "name": "NAME",
                        },
                      },
                    ]
                  }
                  player={
                    Object {
                      "name": "NAME",
                    }
                  }
                />
              </Col>
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={12}
              >
                <Unknown
                  board={
                    Array [
                      "x",
                      "o",
                      "x",
                      null,
                      null,
                      "x",
                      "o",
                      null,
                      null,
                    ]
                  }
                  computerPlay={[Function]}
                  currentPlayer={
                    Object {
                      "name": "NAME",
                    }
                  }
                />
              </Col>
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={12}
              >
                <Unknown
                  history={
                    Array [
                      Object {
                        "id": 1,
                        "winner": Object {
                          "name": "NAME",
                        },
                      },
                    ]
                  }
                />
              </Col>
            </Row>
          </Grid>,
          "className": "content",
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "container",
            "children": <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={12}
              >
                <PiePanel
                  history={
                    Array [
                      Object {
                        "id": 1,
                        "winner": Object {
                          "name": "NAME",
                        },
                      },
                    ]
                  }
                  player={
                    Object {
                      "name": "NAME",
                    }
                  }
                />
              </Col>
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={12}
              >
                <Unknown
                  board={
                    Array [
                      "x",
                      "o",
                      "x",
                      null,
                      null,
                      "x",
                      "o",
                      null,
                      null,
                    ]
                  }
                  computerPlay={[Function]}
                  currentPlayer={
                    Object {
                      "name": "NAME",
                    }
                  }
                />
              </Col>
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={12}
              >
                <Unknown
                  history={
                    Array [
                      Object {
                        "id": 1,
                        "winner": Object {
                          "name": "NAME",
                        },
                      },
                    ]
                  }
                />
              </Col>
            </Row>,
            "componentClass": "div",
            "fluid": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "row",
              "children": Array [
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <PiePanel
                    history={
                      Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ]
                    }
                    player={
                      Object {
                        "name": "NAME",
                      }
                    }
                  />
                </Col>,
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <Unknown
                    board={
                      Array [
                        "x",
                        "o",
                        "x",
                        null,
                        null,
                        "x",
                        "o",
                        null,
                        null,
                      ]
                    }
                    computerPlay={[Function]}
                    currentPlayer={
                      Object {
                        "name": "NAME",
                      }
                    }
                  />
                </Col>,
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <Unknown
                    history={
                      Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ]
                    }
                  />
                </Col>,
              ],
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "col",
                  "children": <PiePanel
                    history={
                      Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ]
                    }
                    player={
                      Object {
                        "name": "NAME",
                      }
                    }
                  />,
                  "componentClass": "div",
                  "md": 4,
                  "xs": 12,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "history": Array [
                      Object {
                        "id": 1,
                        "winner": Object {
                          "name": "NAME",
                        },
                      },
                    ],
                    "player": Object {
                      "name": "NAME",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "col",
                  "children": <Unknown
                    board={
                      Array [
                        "x",
                        "o",
                        "x",
                        null,
                        null,
                        "x",
                        "o",
                        null,
                        null,
                      ]
                    }
                    computerPlay={[Function]}
                    currentPlayer={
                      Object {
                        "name": "NAME",
                      }
                    }
                  />,
                  "componentClass": "div",
                  "md": 4,
                  "xs": 12,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "board": Array [
                      "x",
                      "o",
                      "x",
                      null,
                      null,
                      "x",
                      "o",
                      null,
                      null,
                    ],
                    "computerPlay": [Function],
                    "currentPlayer": Object {
                      "name": "NAME",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "col",
                  "children": <Unknown
                    history={
                      Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ]
                    }
                  />,
                  "componentClass": "div",
                  "md": 4,
                  "xs": 12,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "history": Array [
                      Object {
                        "id": 1,
                        "winner": Object {
                          "name": "NAME",
                        },
                      },
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <Unknown
            player={
              Object {
                "name": "NAME",
              }
            }
          >
            <Unknown>
              <Unknown
                type="trophy"
              />
              <Unknown
                name="TicTacToe"
              />
            </Unknown>
            <Unknown>
              <StartButton
                status="GAME_OVER"
              >
                Start The Game
              </StartButton>
            </Unknown>
          </Unknown>,
          <Jumbotron
            bsClass="jumbotron"
            className="content"
            componentClass="div"
          >
            <Grid
              bsClass="container"
              componentClass="div"
              fluid={false}
            >
              <Row
                bsClass="row"
                componentClass="div"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <PiePanel
                    history={
                      Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ]
                    }
                    player={
                      Object {
                        "name": "NAME",
                      }
                    }
                  />
                </Col>
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <Unknown
                    board={
                      Array [
                        "x",
                        "o",
                        "x",
                        null,
                        null,
                        "x",
                        "o",
                        null,
                        null,
                      ]
                    }
                    computerPlay={[Function]}
                    currentPlayer={
                      Object {
                        "name": "NAME",
                      }
                    }
                  />
                </Col>
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <Unknown
                    history={
                      Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ]
                    }
                  />
                </Col>
              </Row>
            </Grid>
          </Jumbotron>,
        ],
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Unknown>
                <Unknown
                  type="trophy"
                />
                <Unknown
                  name="TicTacToe"
                />
              </Unknown>,
              <Unknown>
                <StartButton
                  status="GAME_OVER"
                >
                  Start The Game
                </StartButton>
              </Unknown>,
            ],
            "player": Object {
              "name": "NAME",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Unknown
                    type="trophy"
                  />,
                  <Unknown
                    name="TicTacToe"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "type": "trophy",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "name": "TicTacToe",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <StartButton
                  status="GAME_OVER"
                >
                  Start The Game
                </StartButton>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Start The Game",
                  "status": "GAME_OVER",
                },
                "ref": null,
                "rendered": "Start The Game",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "jumbotron",
            "children": <Grid
              bsClass="container"
              componentClass="div"
              fluid={false}
            >
              <Row
                bsClass="row"
                componentClass="div"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <PiePanel
                    history={
                      Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ]
                    }
                    player={
                      Object {
                        "name": "NAME",
                      }
                    }
                  />
                </Col>
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <Unknown
                    board={
                      Array [
                        "x",
                        "o",
                        "x",
                        null,
                        null,
                        "x",
                        "o",
                        null,
                        null,
                      ]
                    }
                    computerPlay={[Function]}
                    currentPlayer={
                      Object {
                        "name": "NAME",
                      }
                    }
                  />
                </Col>
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <Unknown
                    history={
                      Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ]
                    }
                  />
                </Col>
              </Row>
            </Grid>,
            "className": "content",
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "container",
              "children": <Row
                bsClass="row"
                componentClass="div"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <PiePanel
                    history={
                      Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ]
                    }
                    player={
                      Object {
                        "name": "NAME",
                      }
                    }
                  />
                </Col>
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <Unknown
                    board={
                      Array [
                        "x",
                        "o",
                        "x",
                        null,
                        null,
                        "x",
                        "o",
                        null,
                        null,
                      ]
                    }
                    computerPlay={[Function]}
                    currentPlayer={
                      Object {
                        "name": "NAME",
                      }
                    }
                  />
                </Col>
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={4}
                  xs={12}
                >
                  <Unknown
                    history={
                      Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ]
                    }
                  />
                </Col>
              </Row>,
              "componentClass": "div",
              "fluid": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "row",
                "children": Array [
                  <Col
                    bsClass="col"
                    componentClass="div"
                    md={4}
                    xs={12}
                  >
                    <PiePanel
                      history={
                        Array [
                          Object {
                            "id": 1,
                            "winner": Object {
                              "name": "NAME",
                            },
                          },
                        ]
                      }
                      player={
                        Object {
                          "name": "NAME",
                        }
                      }
                    />
                  </Col>,
                  <Col
                    bsClass="col"
                    componentClass="div"
                    md={4}
                    xs={12}
                  >
                    <Unknown
                      board={
                        Array [
                          "x",
                          "o",
                          "x",
                          null,
                          null,
                          "x",
                          "o",
                          null,
                          null,
                        ]
                      }
                      computerPlay={[Function]}
                      currentPlayer={
                        Object {
                          "name": "NAME",
                        }
                      }
                    />
                  </Col>,
                  <Col
                    bsClass="col"
                    componentClass="div"
                    md={4}
                    xs={12}
                  >
                    <Unknown
                      history={
                        Array [
                          Object {
                            "id": 1,
                            "winner": Object {
                              "name": "NAME",
                            },
                          },
                        ]
                      }
                    />
                  </Col>,
                ],
                "componentClass": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "col",
                    "children": <PiePanel
                      history={
                        Array [
                          Object {
                            "id": 1,
                            "winner": Object {
                              "name": "NAME",
                            },
                          },
                        ]
                      }
                      player={
                        Object {
                          "name": "NAME",
                        }
                      }
                    />,
                    "componentClass": "div",
                    "md": 4,
                    "xs": 12,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "history": Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ],
                      "player": Object {
                        "name": "NAME",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "col",
                    "children": <Unknown
                      board={
                        Array [
                          "x",
                          "o",
                          "x",
                          null,
                          null,
                          "x",
                          "o",
                          null,
                          null,
                        ]
                      }
                      computerPlay={[Function]}
                      currentPlayer={
                        Object {
                          "name": "NAME",
                        }
                      }
                    />,
                    "componentClass": "div",
                    "md": 4,
                    "xs": 12,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "board": Array [
                        "x",
                        "o",
                        "x",
                        null,
                        null,
                        "x",
                        "o",
                        null,
                        null,
                      ],
                      "computerPlay": [Function],
                      "currentPlayer": Object {
                        "name": "NAME",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "col",
                    "children": <Unknown
                      history={
                        Array [
                          Object {
                            "id": 1,
                            "winner": Object {
                              "name": "NAME",
                            },
                          },
                        ]
                      }
                    />,
                    "componentClass": "div",
                    "md": 4,
                    "xs": 12,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "history": Array [
                        Object {
                          "id": 1,
                          "winner": Object {
                            "name": "NAME",
                          },
                        },
                      ],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
